google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineSupportProvider.java

```diff
@@ -211,42 +212,30 @@ static void removeMavenLibraries(final Set<AppEngineStandardMavenLibrary> librar
     final ModifiableRootModel model = manager.getModifiableModel();
     final LibraryTable libraryTable = ProjectLibraryTable.getInstance(module.getProject());
 
-    ApplicationManager.getApplication().invokeLater(new Runnable() {
-      @Override
-      public void run() {
-        new WriteAction() {
-          @Override
-          protected void run(@NotNull Result result) throws Throwable {
-            for (AppEngineStandardMavenLibrary libraryToRemove : librariesToRemove) {
-              final String displayName = AppEngineStandardMavenLibrary
-                  .toMavenDisplayVersion(libraryToRemove.getLibraryProperties());
-              final Library library = libraryTable.getLibraryByName(displayName);
-              if (library != null) {
-                libraryTable.removeLibrary(library);
-
-                for (OrderEntry orderEntry : model.getOrderEntries()) {
-                  if (orderEntry.getPresentableName().equals(library.getName())) {
-                    model.removeOrderEntry(orderEntry);
-                  }
+    ApplicationManager.getApplication().invokeLater(() -> {
+      new WriteAction() {
```

Comment:
I don't know the answer off hand. runWriteAction does not invoke the Runnable asynchronously so it may be a behavior change. Tt would need to be thoroughly tested.

Regardless, along the lines of Patrick's comment, this change alone could be its own PR.